numpy/distutils/command/build_ext.py

```diff
@@ -54,10 +54,24 @@ def finalize_options(self):
                 self.jobs = int(self.jobs)
             except ValueError:
                 raise ValueError("--jobs/-j argument must be an integer")
-        incl_dirs = self.include_dirs
+
+        # Ensure that self.include_dirs and self.distribution.include_dirs
+        # refer to the same list object. finalize_options will modify
+        # self.include_dirs, but self.distribution.include_dirs is used
+        # during the actual build.
+        # self.include_dirs is None unless paths are specified with
+        # --include-dirs.
+        # The include paths will be passed to the compiler in the order:
+        # numpy paths, --include-dirs paths, Python include path.
+        if isinstance(self.include_dirs, str):
```

Comment:
Sorry for this late comment, but at the moment we have the odd situation where the code modifies `self.include_dirs` and then replaces it.  Would this not be clearer as:

```
        # Ensure include dirs and distribution.include_dirs are lists
        incl_dirs = self.include_dirs or []
        if isinstance(incl_dirs, str):
            incl_dirs = incl_dirs.split(os.pathsep)
        if self.distribution.include_dirs is None:
            self.distribution.include_dirs = []
        # Make sure self.include_dirs is same list as self.distribution.include_dirs
        self.include_dirs = self.distribution.include_dirs
        self.include_dirs.extend(incl_dirs)
```